### ISSUE TITLE
Switching to different app while preparing upload: stuck in queue

### ISSUE THREAD

**Issue by VDK** [commons-app/apps-android-commons#1937]
**Summary:** 
Upload stuck in "queued" when user has switched to a different app before completing the upload process. On rebooting the app, the upload is marked as failed and can't be reïnitiated.

**Steps to reproduce:** 

* select picture to be uploaded
* switch to a different app
* switch back to complete the process
* upload is marked as "queud" but isn't being uploaded. No notification with a loading bar is seen. 
**Device and Android version:** 
Android 8.0.0,  OnePlus's OxygenOS stock ROM, rooted
 **Commons app version:** 
version 2.8.5

**Comment by nicolas-raoul** [comment]
Does it happen 100% of the time when you switch apps?
Does it happen 0% of the time when you do not switch apps?
Thanks in advance! :-)

**Comment by VDK** [comment]
I'm fairly sure it's a 100%/0%
Switching to a different app, to copy the correct spelling of names, is a fairly regular part of my work flow. 
Even so it took a while for me to see the correlation between an upload failing and switching apps.
I stopped doing so when I finally did and since then it hasn't happened that an upload failed.
I tested my hypothesis and even only briefly switching results in the image not being uploaded.
